### ISSUE TITLE
allow "as:" in conjunction with "with:" binding

### ISSUE THREAD

**Issue by joeheyming** [knockout/knockout#1931]
Sometimes, I want to be more explicit than just using $data in a "with:" binding.

If we had an as: modifier to the with: binding, you can create more rich contexts:

&lt;!-- ko with: $parents[2], as: '$descriptiveName' --&gt;
&lt;div data-bind="text: $descriptiveName.property"&gt;&lt;/div&gt;
&lt;!-- /ko --&gt;

This example is not very useful, but I hope you get the idea.

**Comment by mbest** [comment]
Perhaps the discussion in #1792 is helpful.

**Comment by IanYates** [comment]
Yes, this would be nice.  I just augmented my own project with the `withProperties` example from the KO docs as I found something like that handy yesterday.  I'm trying to not rely on $data much at all when I have some nested contexts simply so I can rearrange some parts of the HTML when experimenting without worrying about if I moved a button, and its click/command binding, into a different context.  Less friction => happiness :)
